naga/tests/snapshots.rs

```diff
@@ -775,72 +784,47 @@ fn convert_snapshots_wgsl() {
 }
 
 #[cfg(feature = "spv-in")]
-fn convert_spv(name: &str, adjust_coordinate_space: bool) {
+#[test]
+fn convert_snapshots_spv() {
     use std::process::Command;
 
     let _ = env_logger::try_init();
 
-    let input = Input::new(Some("spv"), name, "spvasm");
-
-    println!("Assembling '{}'", input.file_name.display());
+    for input in Input::files_in_dir(Some("spv"), &["spvasm"]) {
+        println!("Assembling '{}'", input.file_name.display());
 
-    let command = Command::new("spirv-as")
-        .arg(input.input_path())
-        .arg("-o")
-        .arg("-")
-        .output()
-        .expect(
-            "Failed to execute spirv-as. It can be installed \
+        let command = Command::new("spirv-as")
+            .arg(input.input_path())
+            .arg("-o")
+            .arg("-")
+            .output()
+            .expect(
+                "Failed to execute spirv-as. It can be installed \
             by installing the Vulkan SDK and adding it to your path.",
-        );
-
-    println!("Processing '{}'", input.file_name.display());
+            );
 
-    if !command.status.success() {
-        panic!(
-            "spirv-as failed: {}\n{}",
-            String::from_utf8_lossy(&command.stdout),
-            String::from_utf8_lossy(&command.stderr)
-        );
-    }
+        println!("Processing '{}'", input.file_name.display());
 
-    let mut module = naga::front::spv::parse_u8_slice(
-        &command.stdout,
-        &naga::front::spv::Options {
-            adjust_coordinate_space,
-            strict_capabilities: false,
-            block_ctx_dump_prefix: None,
-        },
-    )
-    .unwrap();
+        if !command.status.success() {
+            panic!(
+                "spirv-as failed: {}\n{}",
+                String::from_utf8_lossy(&command.stdout),
+                String::from_utf8_lossy(&command.stderr)
+            );
+        }
 
-    check_targets(&input, &mut module, None);
-}
+        let mut module = naga::front::spv::parse_u8_slice(
+            &command.stdout,
+            &naga::front::spv::Options {
+                adjust_coordinate_space: input.read_parameters().spv_in.adjust_coordinate_space,
```

Comment:
nitpick: I think using a destructuring assignment would be more maintenance-friendly for the future.

Reply:
Good point!